lib/resque/durable.rb

```diff
@@ -66,12 +66,25 @@ def around_perform_manage_audit(*args)
           yield
         end
 
-        a.complete!
+        if graceful_shutdown? # TODO: What if audit is nil?
+          a.re_enqueue! # TODO: will this introduce any kind of race conditions?
+        else
+          a.complete!
+        end
       else
         yield
       end
     end
 
+    def graceful_shutdown?
+      @graceful_shutdown
+    end
+
+    def graceful_shutdown
```

Comment:
who's responsible for calling this?

Reply:
This will be called from https://github.com/zendesk/zendesk/pull/47911/commits/d1d38eb2e27aab56053247d7f365c0e6803d7ff4#diff-e4a2ac7946d691b8c6c0d6d1a9cb14c6d5fab7d7e886e1e3aab09402aa0e5726R103

Whenever worker is trying to exit, we call this method and immediately exit from the job processing loop.